src/openvpn/ring_buffer.h

```diff
@@ -117,7 +119,19 @@ register_ring_buffers(HANDLE device,
 
     res = DeviceIoControl(device, TUN_IOCTL_REGISTER_RINGS, &rr, sizeof(rr),
       NULL, 0, &bytes_returned, NULL);
-
+    /*
+    Sends a control code directly to a specified device driver, causing the corresponding device to perform the corresponding operation.
+    BOOL DeviceIoControl(
+  [in]                HANDLE       hDevice,
+  [in]                DWORD        dwIoControlCode,
+  [in, optional]      LPVOID       lpInBuffer,
+  [in]                DWORD        nInBufferSize,
+  [out, optional]     LPVOID       lpOutBuffer,
+  [in]                DWORD        nOutBufferSize,
+  [out, optional]     LPDWORD      lpBytesReturned,
+  [in, out, optional] LPOVERLAPPED lpOverlapped
+);
```

Comment:
I wonder why we should document library function in OpenVPN. Shouldn't a good IDE also allow you to lookup thiee?

Reply:
The API documentation belongs to where the function is defined/declared (i.e. the SDK), not in the user.